tests/test_release_transpiler_argument_types.py

```diff
@@ -0,0 +1,564 @@
+                VALID_RELEASE_FOLDER,
+                "--release-base",
+                VALID_RELEASE_BASE,
+                "--override-mapping",
+                VALID_OVERRIDE_MAPPING_FILE,
+            ],
+            does_not_raise(),
+        ),
+        (
+            [
+                "combine",
+                "--release-folder",
+                VALID_RELEASE_FOLDER,
+                "--release-base",
+                VALID_RELEASE_BASE,
+                "--override-mapping",
+                "FAKE/PATH/mapping.yml",
+            ],
+            pytest.raises(ARGUMENT_PARSING_ERROR),
+        ),
+        (
+            [
+                "combine",
+                "--release-folder",
+                VALID_RELEASE_FOLDER,
+                "--release-base",
+                VALID_RELEASE_BASE,
+                "--override-mapping",
+                "fake_mapping.yml",
+            ],
+            pytest.raises(ARGUMENT_PARSING_ERROR),
+        ),
+        (
+            [
+                "combine",
+                "--release-folder",
+                VALID_RELEASE_FOLDER,
+                "--release-base",
+                VALID_RELEASE_BASE,
+                "--override-mapping",
+                "random",
+            ],
+            pytest.raises(ARGUMENT_PARSING_ERROR),
+        ),
+        (
+            [
+                "combine",
+                "--release-folder",
+                VALID_RELEASE_FOLDER,
+                "--release-base",
+                VALID_RELEASE_BASE,
+                "--override-mapping",
+                "0",
+            ],
+            pytest.raises(ARGUMENT_PARSING_ERROR),
+        ),
+    ],
+)
+def test_combine_py_override_mapping_type(args: list, expectation, monkeypatch):
+    monkeypatch.setattr(sys, "argv", [""] + args)
+    with expectation:
+        release_transpiler_main()
+
+
+@pytest.mark.parametrize(
+    "args, expectation",
+    [
+        (
+            [
+                "combine",
+                "--release-folder",
+                VALID_RELEASE_FOLDER,
+                "--release-base",
+                VALID_RELEASE_BASE,
+                "--override-mapping",
+                VALID_OVERRIDE_MAPPING_FILE,
+                "--py_coords",
+                "3.8,3.6",
+            ],
+            does_not_raise(),
+        ),
+        (
+            [
+                "combine",
+                "--release-folder",
+                VALID_RELEASE_FOLDER,
+                "--release-base",
+                "INVALID_RELEASE_BASE",
+                "--override-mapping",
+                VALID_OVERRIDE_MAPPING_FILE,
+                "--py_coords",
+                "3.8,3.6",
+            ],
+            pytest.raises(argparse.ArgumentTypeError),
+        ),
+    ],
+)
+def test_combine_py_release_base_type(args: list, expectation, monkeypatch):
+    monkeypatch.setattr(sys, "argv", [""] + args)
+    with expectation:
+        release_transpiler_main()
+
+
+# TESTING TRANSPILER
```

Comment:
I think this comment should go away, what we are really doing should be described by the function name (although sometimes we need to scroll down a bit to read that)